docs/tsp/typeServerProtocol.ts

```diff
@@ -56,20 +56,20 @@ export namespace TypeServerProtocol {
 
     // Represents a category of a type, such as class, function, variable, etc.
     export const enum TypeCategory {
-        Class = 0,
-        Function = 1,
-        Overloaded = 2,
-        Module = 3,
-        Variable = 4,
-        Parameter = 5,
-        Property = 6,
-        TypeAlias = 7,
-        TypeVariable = 8,
-        Generic = 9,
-        Unknown = 10,
-        Any = 11,
-        None = 12,
-        Union = 13,
+        // Type can be anything
+        Any,
+        // Callable type
+        Function,
+        // Functions defined with @overload decorator
+        Overloaded,
+        // Class definition
+        Class,
+        // Module instance
+        Module,
+        // Union of two or more other types
+        Union,
+        // Type variable
+        TypeVar,
     }
 
     // Flags that describe the characteristics of a type.
```

Comment:
cc @ndmitchell

does this make assumptions about how we represent these types that won't be compatible with pyrefly? how will be know if it's missing anything?

Reply:
Not sure if you're asking @ndmitchell or me this question? This set of categories seems pretty generic but if you didn't support them all, you'd just return Any for things you don't have more specific information on.